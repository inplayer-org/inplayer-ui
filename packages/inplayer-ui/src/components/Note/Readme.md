Note is a component for displaying information for notification. <br />
The `type` attribute can be: informative, success, warning, danger. If not of the above is added, the default note with white background will be displayed. <br />
The `title` attribute is the string to be set as title for the notification. <br />
The `text` attribute is the actual notification that needs to be displayed to the user. <br />
The `icon` attribute is a string which represents a FontAwesome icon. It is not considered when there is a `type` prop passed. <br />

You can use it directly like this:

```jsx static
import { Note } from 'inplayer-ui';

function Page(props) {
  return (
    <Note
      type="success"
      title="Done!"
      text="This box indicates a successful or positive action."
    />
  );
}
```

### Demo

```jsx
<Note
  title="Just so you know!"
  text="This block indicates a neutral informative note."
  icon="info-circle"
/>

<Note
  type="informative"
  title="Just so you know!"
  text="This is used as a note on white backgrounds to give it a little contrast."
/>

<Note
  type="success"
  title="Done!"
  text="This box indicates a successful or positive action."
/>

<Note
  type="warning"
  title="Important note!"
  text="This box indicates a warning that might need attention."
/>

<Note
  type="danger"
  title="Danger!"
  text="This box indicates a dangerous or potentially negative action."
/>
```
