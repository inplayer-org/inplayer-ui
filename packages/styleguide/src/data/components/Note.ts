import { ComponentData } from "../types";

const usage = `
  import { Note } from 'inplayer-ui';

  function Page(props) {
    return (
      <Note
        type="success"
        title="Done!"
        text="This box indicates a successful or positive action."
      />
    );
  }
`;

const demoCode = `<div>
  <Note
    title="Just so you know!"
    text="This block indicates a neutral informative note."
    icon="info-circle"
  />

  <Note
    type="informative"
    title="Just so you know!"
    text="This is used as a note on white backgrounds to give it a little contrast."
  />

  <Note
    type="success"
    title="Done!"
    text="This box indicates a successful or positive action."
  />

  <Note
    type="warning"
    title="Important note!"
    text="This box indicates a warning that might need attention."
  />

  <Note
    type="danger"
    title="Danger!"
    text="This box indicates a dangerous or potentially negative action."
  />
</div>
`;

const Note: ComponentData = {
  description: `Note is a component for displaying information for notification. 
  The type attribute can be: informative, success, warning, danger. If not of the above is added, the default note with white background will be displayed. 
  The title attribute is the string to be set as title for the notification. 
  The text attribute is the actual notification that needs to be displayed to the user. 
  The icon attribute is a string which represents a FontAwesome icon. It is not considered when there is a type prop passed.`,
  path: 'src/components/Note/index.js',
  usage,
  examples: [
    {
      title: 'Demo',
      code: demoCode,
    }
  ],
  propsAndMethods: [
    { name: 'icon', type: 'string', default: 'Required' },
    { name: 'text', type: 'string', default: 'Required' },
    { name: 'title', type: 'string', default: 'Required' },
    { name: 'type', type: 'enum', default: 'Required', description: 'One of: informative, success, warning, danger' },
    { name: 'className', type: 'string', description: 'A className can be passed down for further styling or extending with CSS-in-JS' },
    { name: 'style', type: 'Object', default: '{ }' },
  ],
}

export default Note;