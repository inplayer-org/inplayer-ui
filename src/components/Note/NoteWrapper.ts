import styled, { css } from 'styled-components';
import { switchProp } from 'styled-tools';

import colors from '../../theme/colors';

export type NoteType = 'informative' | 'success' | 'warning' | 'danger';

const NoteWrapper = styled.p<{ type: NoteType }>`
  display: flex;
  align-items: center;
  box-sizing: border-box;
  border: 1px solid ${colors.gray};
  padding: 1.125rem 2%;
  border-radius: 3px;
  background: ${colors.white};
  color: ${({ theme }) => theme.palette.text.main};
  font-weight: ${({ theme }) => theme.font.weights.light};
  letter-spacing: 0.01em;
  line-height: 1.8;

  strong {
    padding-right: 0.3125rem;
  }

  i {
    margin-right: 0.75rem;
    position: relative;
    top: 2px;
  }

  ${switchProp('type', {
    success: css`
      border-color: ${colors.skyBlue};

      strong {
        color: ${colors.skyBlue};
      }

      i {
        color: ${colors.skyBlue};
      }
    `,

    warning: css`
      border-color: ${colors.yellow};
      background: ${colors.paleYellow};

      i {
        color: ${colors.yellow};
      }
    `,

    danger: css`
      border-color: ${colors.red};
      background: ${colors.paleRed};

      i {
        color: ${colors.red};
      }
    `,

    informative: css`
      background: ${colors.lightGray};
    `,
  })};
`;

export default NoteWrapper;
