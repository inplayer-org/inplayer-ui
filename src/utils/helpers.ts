import moment from 'moment';

export const getMonthOptions = (): Array<any> =>
  moment.months().map((label, value) => ({ value, displayName: label }));

export const getYearOptions = () => {
  const yearOptions: Array<any> = [];
  for (let i = moment().year() - 100; i <= moment().year(); i += 1) {
    yearOptions.push({ value: i.toString(), displayName: i.toString() });
  }

  return yearOptions;
};
