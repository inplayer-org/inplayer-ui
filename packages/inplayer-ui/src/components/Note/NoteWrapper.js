import styled, { css } from 'styled-components';
import colors from 'config/colors';
import { switchProp } from 'styled-tools';
import { fontWeights, uiColors } from 'utils';

const NoteWrapper = styled.p`
  box-sizing: border-box;
  border: 1px solid ${colors.gray};
  padding: 1.125rem 2%;
  border-radius: 3px;
  background: ${colors.white};
  color: ${uiColors('text.main')};
  font-weight: ${fontWeights('light')};
  letter-spacing: 0.01em;
  line-height: 1.8;

  strong {
    padding-right: 0.3125rem;
  }

  i {
    margin-right: 0.75rem;
    position: relative;
    top: 2px;
  }

  ${switchProp('type', {
    success: css`
      border-color: ${colors.skyBlue};

      strong {
        color: ${colors.skyBlue};
      }

      i {
        color: ${colors.skyBlue};
      }
    `,

    warning: css`
      border-color: ${colors.yellow};
      background: ${colors.paleYellow};

      i {
        color: ${colors.yellow};
      }
    `,

    danger: css`
      border-color: ${colors.red};
      background: ${colors.paleRed};

      i {
        color: ${colors.red};
      }
    `,

    informative: css`
      background: ${colors.lightGray};
    `,
  })};
`;

export default NoteWrapper;
