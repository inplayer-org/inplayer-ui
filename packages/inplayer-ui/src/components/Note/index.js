// @flow
import React from 'react';
import Icon from 'elements/Icon';
import NoteWrapper from './NoteWrapper';

type NoteType = 'informative' | 'success' | 'warning' | 'danger';

type Props = {
  title: string,
  text: string,
  icon: string,
  type: NoteType,
  /** A className can be passed down for further styling or extending with CSS-in-JS */
  className?: string,
  style?: Object,
};

const NoteIcon = {
  informative: null,
  success: 'check-circle',
  warning: 'info-circle',
  danger: 'exclamation-triangle',
};

const NoteIconModifier = {
  informative: null,
  success: 'statusColorPrimary',
  warning: 'statusColorWarning',
  danger: 'statusColorDanger',
};

const getIconFromNoteType = (icon: string, type: NoteType) =>
  type !== 'informative' && (NoteIcon[type] || icon);

const Note = ({ type, title, text, icon, className, style }: Props) => {
  const iconType = getIconFromNoteType(icon, type);
  return (
    <NoteWrapper type={type} className={className} style={style}>
      {iconType && (
        <Icon name={getIconFromNoteType(icon, type)} modifiers={[NoteIconModifier[type]]} />
      )}{' '}
      <strong>{title}</strong>
      {text}
    </NoteWrapper>
  );
};

Note.defaultProps = {
  className: '',
  style: {},
};

export default Note;
