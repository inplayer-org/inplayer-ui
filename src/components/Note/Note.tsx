import React from 'react';
import styled from 'styled-components';
import { MdInfoOutline, MdWarning } from 'react-icons/md';
import { IoIosCheckmarkCircle } from 'react-icons/io';

import colors from '../../theme/colors';
import { AnalyticsProps } from '../../analytics';

import NoteWrapper, { NoteType } from './NoteWrapper';

type Props = {
  title: string;
  text: string;
  type: NoteType;
} & AnalyticsProps;

interface IconProps {
  color?: string;
}

const createIconStyles = ({ color }: IconProps) => ({
  width: '1.5rem',
  height: '1.5rem',
  color,
  'margin-right': '0.5rem',
});

const SuccessIcon = styled(IoIosCheckmarkCircle)`
  ${createIconStyles}
`;

const WarningIcon = styled(MdInfoOutline)`
  ${createIconStyles}
`;

const DangerIcon = styled(MdWarning)`
  ${createIconStyles}
`;

const NoteIcon = {
  informative: null,
  success: <SuccessIcon color={colors.skyBlue} />,
  warning: <WarningIcon color={colors.yellow} />,
  danger: <DangerIcon color={colors.red} />,
};

const getIconFromNoteType = (type: NoteType) => NoteIcon[type];

const Note = ({ type, title, text }: Props) => {
  const Icon = getIconFromNoteType(type);
  return (
    <NoteWrapper type={type}>
      {Icon} <strong>{title}</strong>
      {text}
    </NoteWrapper>
  );
};

export default Note;
