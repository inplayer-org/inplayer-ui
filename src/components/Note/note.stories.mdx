import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Note from './'

<Meta title="Note" component={<Note/>} />

# Note

Note is a component for displaying information for notification. <br/>
The type attribute can be: `informative`, `success`, `warning`, `danger`. If not of the above is added, the `default` note with white background will be displayed.

```jsx
import { Note } from '@inplayer-org/inplayer-ui';

const Page = () => (
  <Note />
)
```
### Props 

<Props of={Note}/>

### Default
<Preview withSource="open">
  <Story name="default">
    <Note
        title="Just so you know!"
        text="This block indicates a neutral informative note."
        icon="info-circle"
    />
  </Story>
</Preview>

### Informative
<Preview withSource="open">
  <Story name="informative">
    <Note
        type="informative"
        title="Just so you know!"
        text="This is used as a note on white backgrounds to give it a little contrast."
    />
  </Story>
</Preview>

### Success
<Preview withSource="open">
  <Story name="success">
    <Note
        type="success"
        title="Done!"
        text="This box indicates a successful or positive action."
    />
  </Story>
</Preview>

### Warning
<Preview withSource="open">
  <Story name="warning">
    <Note
        type="warning"
        title="Important note!"
        text="This box indicates a warning that might need attention."
    />
  </Story>
</Preview>

### Danger
<Preview withSource="open">
  <Story name="danger">
    <Note
        type="danger"
        title="Danger!"
        text="This box indicates a dangerous or potentially negative action."
    />
  </Story>
</Preview>

