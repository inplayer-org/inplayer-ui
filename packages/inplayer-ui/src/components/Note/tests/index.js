import { generateSnapshots } from 'packages/inplayer-ui/src/components/Note/tests/helpers';

import Note from '../index';

generateSnapshots(Note);
