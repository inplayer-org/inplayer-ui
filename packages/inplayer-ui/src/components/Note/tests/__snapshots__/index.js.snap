// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders correctly 1`] = `
.c0 {
  box-sizing: border-box;
  border: 1px solid #d7dde5;
  padding: 1.125rem 2%;
  border-radius: 3px;
  background: #fff;
  color: #3e4348;
  font-weight: 300;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  line-height: 1.8;
}

.c0 strong {
  padding-right: 0.3125rem;
}

.c0 i {
  margin-right: 0.75rem;
  position: relative;
  top: 2px;
}

<p
  className="c0"
  style={Object {}}
>
   
  <strong />
</p>
`;
